src/segmented_buffer.rs

```diff
@@ -728,10 +741,22 @@ mod test {
         fn write_to_segmented_bool_buf(
             inp in (0..100*1024usize)
                 .prop_flat_map(|size|(Just(size),
-                                      proptest::collection::vec(proptest::num::u8::ANY, size)))){
+                                      proptest::collection::vec(proptest::num::u8::ANY, size)))) {
+
+            countme::enable(true);
+
+            let counts = countme::get::<SegmentedBuf<Reusable<Buffer>>>();
+
+            // Ensure we havn't allocated any bufs yet
+            assert_eq!(counts.live, 0);
+
             let mut buf = SegmentedPoolBufBuilder::new().segment_size(2048).initial_capacity(8192).build();
+
+            // Ensure we havn't allocated more bufs than necessary
+            assert!(counts.live < 8192/2048);
```

Comment:
I would expect to be exactly `4` in this case, not less than. Just curious, whats the reason?

Reply:
This was broken, I hadn't understood that counts.live was populated when you call countme::get rather than being a pointer. I fixed it in the other test but hadn't noticed it was still broken here. It's gone anyway :)